src/js/msp/MSPHelper.js

```diff
@@ -1783,7 +1783,14 @@ MspHelper.prototype.dataflashRead = function(address, blockSize, onDataCallback)
                  * figure out the reply format:
                  */
                 if (dataCompressionType == 0) {
-                    onDataCallback(address, new DataView(response.data.buffer, response.data.byteOffset + headerSize, dataSize));
+
+                    var offset = response.data.byteOffset + headerSize;
+                    var remainingDataSize = response.data.byteLength - offset;
+                    if (dataSize > remainingDataSize) {
```

Comment:
With the expected discrepancy between actual length and `dataSize` removed in betaflight/betaflight#7181, I think this is not the behaviour we want - if the declared and actual length does not match there is an external issue (e.g. data lost in transit), and we should error out instead of accepting the potentially corrupted data and create a corrupted log file.

Reply:
that's not what happens in practice.  when current firmware that sends a smaller amount than requested the file is not corrupted because the next request continues where it should.  I verified this on a 128MB flash while where every single request for 4096 bytes resulted on 2048 bytes being returned.  The file that was downloaded was not corrupt.

Reply:
if the data was truncated or expanded the packet CRC would not be in the correct location or correct.  if the CRC is correct the data is OK.  without the CRC protecting the data i would agree with you, but we have it and use it before the above code is ever reached.

Reply:
Ok, in this case checking `dataSize` should be abandoned as it is devoid of any value, and we should be changing the firmware to send 0 for `dataSize` as sending a random value is only confusing.

Reply:
yeah, indeed. I guess we could just ignore it completely.  I'll try and find some time to update this PR soon.  Bit busy at the moment.

Do you want a new PR for the firmware, I think it makes sense as we can delete some code there to free a bit of flash space if we always send `0` for the data size.